src/main/resources/xsl/sarif.xsl

```diff
@@ -573,87 +573,124 @@
-                                                <xsl:value-of select="concat('/', $locNode/@resProjPath)"/><xsl:text>"</xsl:text>
-                                            </xsl:otherwise>
-                                        </xsl:choose>
+                                    <xsl:when test="$isAnyMatched">
+                                        <!-- Use relative uri when the uri has matching project root path -->
+                                        <xsl:call-template name="relative_artifact_location">
+                                            <xsl:with-param name="uri" select="substring-after($locNode/@uri, $matchedProjectRootPath/RESULT/@uri)"/>
+                                            <xsl:with-param name="uriBaseId" select="$matchedProjectRootPath/RESULT/@name"/>
+                                        </xsl:call-template>
                                     </xsl:when>
                                     <xsl:otherwise>
-                                        <xsl:value-of select="@locFile"/><xsl:text>"</xsl:text>
+                                        <!-- Use default uri when the uri doesn't have matching project root path -->
+                                        <xsl:call-template name="default_artifact_location">
+                                            <xsl:with-param name="uri" select="$locNode/@uri"/>
+                                        </xsl:call-template>
                                     </xsl:otherwise>
                                 </xsl:choose>
                             </xsl:otherwise>
                         </xsl:choose>
                     </xsl:when>
                     <xsl:otherwise>
-                        <xsl:value-of select="@locFile"/><xsl:text>"</xsl:text>
+                        <!-- No <Loc> matches with current violation -->
+                        <xsl:call-template name="default_artifact_location">
+                            <xsl:with-param name="uri" select="@locFile"/>
+                        </xsl:call-template>
                     </xsl:otherwise>
                 </xsl:choose>
-             </xsl:otherwise>
+            </xsl:otherwise>
         </xsl:choose>
 
         <xsl:text> }</xsl:text>
     </xsl:template>
 
-    <xsl:template name="getMatchedProjectRootPath">
+    <xsl:template name="default_artifact_location">
+        <xsl:param name="uri"/>
+        <xsl:text>"uri": " </xsl:text>
+        <xsl:value-of select="$uri"/>
+        <xsl:text>"</xsl:text>
+    </xsl:template>
+
+    <xsl:template name="relative_artifact_location">
+        <xsl:param name="uri"/>
+        <xsl:param name="uriBaseId"/>
+        <xsl:text>"uri": "</xsl:text>
+        <xsl:value-of select="$uri"/>
+        <xsl:text>", "uriBaseId": "</xsl:text>
+        <xsl:value-of select="$uriBaseId"/>
+        <xsl:text>"</xsl:text>
+    </xsl:template>
+
+    <xsl:template name="getMatchedProjectRootPath_by_loc">
```

Comment:
Should we keep this template? Since it is unused in this file.

Reply:
You're right , it's not used anymore, I will remove it.